neurom/core/_neuron.py

```diff
@@ -152,6 +154,43 @@ def graft_neuron(root_section):
     return Neuron(soma=Soma(root_section.points[:1]), neurites=[Neurite(root_section)])
 
 
+def sample_morph_points(morph, sample_distance):
```

Comment:
if it will accept a neurite type as an argument, maybe we should rename it to iter_positions and have it return an iterator? This may be a clearer name, and more consistent with the other methods exposed at the top-level of neurom.

Reply:
if we do want this consistency, maybe making it accept a filter like the other iterators would be preferable.

Reply:
I think the iterator is a good idea. However I am a bit confused on what's the best option. What I had in mind was to implement something like ```iter_segments(obj, neurite_filter=None, neurite_order=NeuriteIter.FileOrder)``` to replace the segments that are returned according to a specific sampling. However, as I see from the code that would be hard to do because you do not iterate over existing points, but you generate new points. 

Another proposal would be to take as input a neurite and return a resampled neurite. This can then be used for any additional analysis as a regular neurite. I'm not sure what's the best option. 

It would be helpful to give us your input @wizmer @mgeplf

Reply:
I think we have different ideas on what is needed from this function.
On my end:
To extract the needed biodata you need the y-positions of points every x um along the neurites to convert to relative depth.
To analyse the in-situ synthesized morphologies I need the positions of points every x um along the neurites to index a VoxelData containing relative depth.